std/range/package.d

```diff
@@ -5747,6 +5751,18 @@ debug @system unittest
     }
 }
 
+@safe @nogc nothrow unittest
+{
+    {
+        ushort start = 0, end = 10, step = 2;
+        foreach (i; iota(start, end, step)) {}
```

Comment:
It may be useful to statically assert that `i` is of the expected type -`ushort` here and `ubyte` below.